sdk/eventhub/azure-eventhub/tests/livetest/synctests/test_consumer_client.py

```diff
@@ -103,3 +103,89 @@ def on_event(partition_context, event):
         time.sleep(10)
         assert len(client1._event_processors[("$default", ALL_PARTITIONS)]._consumers) == 1
         assert len(client2._event_processors[("$default", ALL_PARTITIONS)]._consumers) == 1
+
+
+def test_receive_batch_no_max_wait_time(connstr_senders):
+    '''Test whether callback is called when max_wait_time is None and max_batch_size has reached
+    '''
+    connection_str, senders = connstr_senders
+    senders[0].send(EventData("Test EventData"))
+    senders[1].send(EventData("Test EventData"))
+    client = EventHubConsumerClient.from_connection_string(connection_str, consumer_group='$default')
+
+    def on_event_batch(partition_context, event_batch):
+        on_event_batch.received += len(event_batch)
+        partition_context.update_checkpoint()
+        on_event_batch.namespace = partition_context.fully_qualified_namespace
+        on_event_batch.eventhub_name = partition_context.eventhub_name
+        on_event_batch.consumer_group = partition_context.consumer_group
+        on_event_batch.offset = event_batch[-1].offset
+        on_event_batch.sequence_number = event_batch[-1].sequence_number
+
+    on_event_batch.received = 0
+    on_event_batch.namespace = None
+    on_event_batch.eventhub_name = None
+    on_event_batch.consumer_group = None
+    on_event_batch.offset = None
+    on_event_batch.sequence_number = None
+
+    with client:
+        worker = threading.Thread(target=client.receive_batch, args=(on_event_batch,),
+                                  kwargs={"starting_position": "-1"})
+        worker.start()
+        time.sleep(10)
+        assert on_event_batch.received == 2
+
+        checkpoints = list(client._event_processors.values())[0]._checkpoint_store.list_checkpoints(
+            on_event_batch.namespace, on_event_batch.eventhub_name, on_event_batch.consumer_group
+        )
+        assert len([checkpoint for checkpoint in checkpoints if checkpoint["offset"] == on_event_batch.offset]) > 0
+        assert len(
+            [checkpoint for checkpoint in checkpoints if checkpoint["sequence_number"] == on_event_batch.sequence_number]) > 0
+
+    worker.join()
+
+
+@pytest.mark.parametrize("max_wait_time, sleep_time, expected_result",
+                         [(3, 6, []),
+                          (3, 2, None),
```

Comment:
Why is the callback called with None here?

Reply:
Ah - I see it's because we're starting with None and the callback is never called?